### PR TITLE
Removing Dispose from Transport, Connection and HubConnection

### PR DESCRIPTION
**Pull request by moozzyk**
Converting StopAsync to DisposeAsync

### CODE REVIEW

File: src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs [hunk 1]
```diff
@@ -38,12 +38,8 @@ public WebSocketsTransport(ILoggerFactory loggerFactory)
                 throw new ArgumentNullException(nameof(url));
             }
 
-            if (application == null)
-            {
-                throw new ArgumentNullException(nameof(application));
-            }
+            _application = application ?? throw new ArgumentNullException(nameof(application));
```
Comment (mikaelm12) [inline src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs]: Do we want to move to this pattern for param checking?
It's not consistent with how we do it right above

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs]: You can't do this above because url is not being assigned to anything.

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs]: We're not using throw expressions. Get rid of the,.

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs]: Why? What's wrong with them?

**Comment by moozzyk** [comment]
⌚️

File: src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs (hunk 1 above)
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs]: if (... == null) looks better IMO

**Comment by BrennanConroy** [comment]
Looks good other than the `?? throw`

**Comment by anurse** [comment]
Woo! `DisposeAsync`. Now we just need a syntax feature ;)

**Pull request merged**